processing/src/main/java/io/druid/query/groupby/GroupByQueryEngine.java

```diff
@@ -148,7 +148,8 @@ public void close() throws IOException
     private final BufferAggregator[] aggregators;
     private final PositionMaintainer positionMaintainer;
 
-    private final TreeMap<ByteBuffer, Integer> positions;
+    private final Map<ByteBuffer, Integer> positions = Maps.newTreeMap();
```

Comment:
Looking at the code, it seems like a treemap isn't required and we can use a hashmap for positions and avoid the positionsHash variable